src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,127 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<NewMovieProps> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+  // тут !'' = true значить кнопка disabled
+  const notFilled = !newMovie.title || !newMovie.imgUrl
+    || !newMovie.imdbUrl || !newMovie.imdbId;
+
+  const isValidImgUrl = pattern.test(newMovie.imgUrl);
+  const isValidImdbUrl = pattern.test(newMovie.imdbUrl);
+
+  const handleFieldChange = (fieldName: string, value: string) => {
+    setNewMovie(prevState => ({
+      ...prevState,
+      [fieldName]: value,
+    }));
+  };
+
+  const clearForm = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd(newMovie);
+    clearForm();
+    setCount(prevCount => prevCount + 1);
+    setButtonDisabled(true);
+  };
+
+  const checkFields = () => {
+    if ((notFilled === false)
+      && (isValidImgUrl === true) && (isValidImdbUrl === true)) {
```

Comment:
Use logical operators. It'll significantly simplify your code 
```suggestion
    if (!notFilled && isValidImgUrl && isValidImdbUrl) {
```